ui/pages/confirmations/components/confirm/info/typed-sign/typed-sign-permission.tsx

```diff
@@ -0,0 +1,448 @@
+import {
+  Erc20TokenPeriodicPermission,
+  Erc20TokenStreamPermission,
+  NativeTokenPeriodicPermission,
+  NativeTokenStreamPermission,
+} from '@metamask/gator-permissions-controller';
+import { hexToBigInt } from '@metamask/utils';
+import React from 'react';
+import { isSnapId } from '@metamask/snaps-utils';
+
+import { SignatureRequestType } from '../../../../types/confirm';
+import { useConfirmContext } from '../../../../context/confirm';
+import {
+  ConfirmInfoRow,
+  ConfirmInfoRowDate,
+  ConfirmInfoRowDivider,
+} from '../../../../../../components/app/confirm/info/row';
+import { ConfirmInfoSection } from '../../../../../../components/app/confirm/info/row/section';
+import { ConfirmInfoRowUrl } from '../../../../../../components/app/confirm/info/row/url';
+import { useI18nContext } from '../../../../../../hooks/useI18nContext';
+import { NetworkRow } from '../shared/network-row/network-row';
+import { ConfirmInfoAlertRow } from '../../../../../../components/app/confirm/info/row/alert-row/alert-row';
+import { RowAlertKey } from '../../../../../../components/app/confirm/info/row/constants';
+import { SigningInWithRow } from '../shared/sign-in-with-row/sign-in-with-row';
+import { ConfirmInfoRowCurrency } from '../../../../../../components/app/confirm/info/row/currency';
+import { DAY, WEEK } from '../../../../../../../shared/constants/time';
+
+/**
+ * Formats a period duration in seconds to a human-readable string.
+ * Converts common durations (daily, weekly) to readable labels, otherwise shows seconds.
+ *
+ * @param seconds - The duration in seconds to format
+ * @returns A formatted string representing the duration (e.g., "Daily", "Weekly", "3600 seconds")
+ */
+const formatPeriodDuration = (seconds: number) => {
+  // multiply by 1000 to convert to milliseconds
+  switch (seconds * 1000) {
+    case DAY:
+      return 'Daily';
+    case WEEK:
+      return 'Weekly';
+    default:
+      return `${seconds} seconds`;
```

Comment:
This could be updated to show seconds/minutes/days/weeks whichever is more appropriate based on the amount. So that it doesn't say 120 seconds but 2 minutes. But can be done in a separate PR.

Reply:
Yes! I think we should leave this until we do an actual designed UI.